chains/orchestrator-relays/runtime/dancelight/src/bridge_to_ethereum_config.rs

```diff
@@ -345,6 +333,269 @@ where
+            Destination::ForeignAccountId32 { para_id, id, fee } => {
+                let beneficiary = Location::new(0, [AccountId32 { network: None, id }]);
+                Some((beneficiary, fee, para_id))
+            }
+            Destination::ForeignAccountId20 { para_id, id, fee } => {
+                let beneficiary = Location::new(
+                    0,
+                    [AccountKey20 {
+                        network: None,
+                        key: id,
+                    }],
+                );
+                Some((beneficiary, fee, para_id))
+            }
+            _ => {
+                log::error!("NativeContainerTokensProcessor::process_native_token_transfer: invalid destination");
+                None
+            }
+        };
+
+        if let Some((beneficiary, container_fee, container_para_id)) = destination_info {
+            let network = EthereumNetwork::get();
+            let bridge_location = Location::new(2, GlobalConsensus(network));
+
+            let total_fees = token_data.fee.saturating_add(container_fee);
+
+            let container_location = Location::new(0, [Parachain(container_para_id)]);
+
+            let token_split = token_location.interior().clone().split_global().ok();
+
+            if let Some((_, interior)) = token_split {
+                let container_token_from_tanssi = Location::new(0, interior);
+                let reanchor_result = container_token_from_tanssi.reanchored(
+                    &container_location,
+                    &<T as pallet_xcm::Config>::UniversalLocation::get(),
+                );
+
+                if let Ok(token_location_reanchored) = reanchor_result {
+                    let container_asset: Asset =
+                        (token_location_reanchored, token_data.amount).into();
+                    let tanssi_asset_fee: Asset = (Location::parent(), total_fees).into();
+                    let inbound_queue_pallet_index = InboundQueuePalletInstance::get();
+
+                    let remote_xcm = Xcm::<()>(vec![
+                        DescendOrigin(PalletInstance(inbound_queue_pallet_index).into()),
+                        UniversalOrigin(GlobalConsensus(network)),
+                        WithdrawAsset(
+                            vec![tanssi_asset_fee.clone(), container_asset.clone()].into(),
+                        ),
+                        BuyExecution {
+                            fees: tanssi_asset_fee,
+                            weight_limit: Unlimited,
+                        },
+                        DepositAsset {
+                            assets: Definite(container_asset.into()),
+                            beneficiary,
+                        },
+                        // When the execution finishes deposit any leftover assets to the ETH
+                        // sovereign account on destination.
+                        SetAppendix(Xcm(vec![DepositAsset {
```

Comment:
Maybe we need to leave the user tokens trapped, so that it can be claimed later on by the user in case of failed execution of xcm.

Reply:
@Agusrodri Yeah. Not forever. But, will require manual intervention to check the asset amount.